agents/go-agents/exec-agent/main.go

```diff
@@ -105,14 +119,15 @@ func main() {
 	log.Fatal(server.ListenAndServe())
 }
 
-func getHandler(h http.Handler) http.Handler {
-	// required authentication for all the requests, if it is configured
-	if config.authEnabled {
-		cache := auth.NewCache(time.Minute*time.Duration(config.tokensExpirationTimeoutInMinutes), time.Minute*5)
-		return auth.NewCachingHandler(h, config.apiEndpoint, droppingRPCChannelsUnauthorizedHandler, cache)
+func wrapWithAuth(h http.Handler, mapping string) http.Handler {
+	// required authentication for all the requests that match mappings, if auth is configured
+	if !config.authEnabled {
```

Comment:
Maybe it would be better to place here checking of `noAuthPattern` instead of putting it into `NewCachingHandler`

Reply:
I didn't get what you mean. Can you elaborate on that?

Reply:
I mean rewriting it in the following way:
```go

func wrapWithAuth(h http.Handler, ignoreMapping string) http.Handler {
	// required authentication for all the requests that match mappings, if auth is configured
	noAuthPattern := regexp.MustCompile(ignoreMapping)
	if !config.authEnabled || handler.noAuthPattern.MatchString(req.URL.Path){
		return h
	}

	cache := auth.NewCache(time.Minute*time.Duration(config.tokensExpirationTimeoutInMinutes), time.Minute*5)
	return auth.NewCachingHandler(h, config.apiEndpoint, droppingRPCChannelsUnauthorizedHandler, cache)
}
```
It's just proposal. Maybe it's not better. Why I think about it because your comments says
```
	// required authentication for all the requests that match mappings, if auth is configured
```
but actually, it doesn't check mapping at all.
The second thing is if CachingHandler checks mapping, why it doesn't check config?

Reply:
We can check mapping on a per-request basis only. Because of that, we have to check it as part of the auth mechanism. To avoid that we would need a way how to map auth filtering to certain paths, but we have an only single mapping. 
To make it prettier it is needed to rewrite routing mechanism to add auth there. I believe that it is not the right time to do it now. 
checking that auth is enabled is needed here because in the case it is not we don't wrap routing with auth at all.

Reply:
OK then